From 117f0c39091e1c03d4d64b1177693bf952c81c97 Mon Sep 17 00:00:00 2001
From: Patrick Mooney <pmooney@pfmooney.com>
Date: Tue, 11 Oct 2016 19:53:13 +0000
Subject: [PATCH] OS-5711 lxbrand panic in sched_getaffinity Reviewed by: Jerry
 Jelinek <jerry.jelinek@joyent.com> Approved by: Robert Mustacchi
 <rm@joyent.com>

---
 usr/src/uts/common/brand/lx/syscall/lx_sched.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr/src/uts/common/brand/lx/syscall/lx_sched.c b/usr/src/uts/common/brand/lx/syscall/lx_sched.c
index 14d2eda4ea..7607166722 100644
--- a/usr/src/uts/common/brand/lx/syscall/lx_sched.c
+++ b/usr/src/uts/common/brand/lx/syscall/lx_sched.c
@@ -123,7 +123,7 @@ lx_sched_pidlock(l_pid_t pid, proc_t **pp, kthread_t **tp, boolean_t is_write)
 		mutex_exit(&p->p_lock);
 		return (ESRCH);
 	} else if (err > 0) {
-		sprwaitlock_proc(p);
+		sprlock_proc(p);
 		err = 0;
 	}
 
-- 
2.21.0

